src/execution_plan/ops/op_node_by_id_seek.c

```diff
@@ -0,0 +1,97 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "op_node_by_id_seek.h"
+
+#define MIN(a,b) (((a) < (b)) ? (a) : (b))
+#define MAX(a,b) (((a) < (b)) ? (b) : (a))
+
+// Checks to see if operation index is within its bounds.
+static inline bool _outOfBounds(OpNodeByIdSeek *op) {
+    /* Because currentId starts at minimum and only increases
+     * we only care about top bound. */
+    if(op->currentId > op->maxId) return true;
+    if(op->currentId == op->maxId && !op->maxExclusive) return true;
+    return false;
+}
+
+OpBase* NewOpNodeByIdSeekOp
+(
+    const AST *ast,
+    unsigned int nodeRecIdx,
+    NodeID minId,
+    NodeID maxId,
+    bool minExclusive,
+    bool maxExclusive
+)
+{
+    // Can't include unspecified bound.
+    assert(!(minId == ID_RANGE_UNBOUND && minExclusive));
+    assert(!(maxId == ID_RANGE_UNBOUND && maxExclusive));
+
+    OpNodeByIdSeek *op_nodeByIdSeek = malloc(sizeof(OpNodeByIdSeek));
+    op_nodeByIdSeek->g = GraphContext_GetFromTLS()->g;
+    
+    op_nodeByIdSeek->minExclusive = minExclusive;
+    op_nodeByIdSeek->maxExclusive = maxExclusive;
+
+    // The smallest possible entity ID is 0.
+    op_nodeByIdSeek->minId = MAX(0, minId);
```

Comment:
This doesn't work, since `NodeID` is unsigned - `ID_RANGE_UNBOUND` is actually UINT64_MAX.
Instead, check explicitly for `if (minId == ID_RANGE_UNBOUND)`.